### DISCUSSION TITLE
AttributeError after upgrading ibis

### DISCUSSION THREAD

**Discussion by vmsaipreeth** [ibis-project/ibis#10068] [Q&A] [answered]
Hi Team,

Below is the sample code where we connect to trino via ibis and fetch data.

```
conn = ibis.trino.connect(
    host=‘example host’,
    user=‘username’,
    password=‘password’,
    port=port,
    http_schema=‘https’,
    database=‘database’,
    schema=‘schema’)

data = conn.sql(“SQL Query”)
```

The above code works fine with `ibis-framework == 8.0.0`

Recently, we upgraded to `ibis-framework == 9.4.0`
And noticed an `AttributeError: ‘str’ object has no attribute ’set_http_session’` at `conn.sql(“SQL Query”) `

Trino version: `0.329.0`

Please let me know if anything got changed here.

TIA!

**Comment by gforsyth** [comment, marked as answer]
This is fixed on `main` and should be released in 9.5.0 any day now, but as a workaround, you can do:

```python
from trino.auth import BasicAuthentication

conn = ibis.trino.connect(
    host=‘example host’,
    user=‘username’,
    port=port,
    auth=BasicAuthentication("username", "password"),
    http_schema=‘https’,
    database=‘database’,
    schema=‘schema’)

**Reply by cpcloud** [reply]
This should've been fixed in 9.4.0 though

https://github.com/ibis-project/ibis/pull/9960

**Reply by cpcloud** [reply]
I think there's an issue with our conditional, and the inspection of the `scheme` in the parsed `host` URL.

It looks like the trino client allows you to pass a separate kwarg for the http scheme, which is horrible.

We should maybe forget about trying to parse the host.

**Reply by vmsaipreeth** [reply]
The work around provided by @gforsyth works in `9.4.0`. Thanks!
